usecases/schema/property.go

```diff
@@ -16,47 +16,92 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/pkg/errors"
 	"github.com/weaviate/weaviate/entities/models"
 	"github.com/weaviate/weaviate/entities/schema"
 )
 
-var errPropertyNotFound = errors.New("property not found")
-
-// AddClassProperty to an existing Class
+// AddClassProperty it is upsert. it adds properties to a class and updates
+// existing properties if the merge bool passed true.
 func (h *Handler) AddClassProperty(ctx context.Context, principal *models.Principal,
-	class string, prop *models.Property,
+	class *models.Class, merge bool, newProps ...*models.Property,
 ) error {
+	if len(newProps) == 0 {
```

Comment:
question: should we move that check after the authorizer check ? I'm always a bit afraid we might extend that code with more in the future and forget that the authorizer check is afterward. A bit of defensive programming IMO we shouldn't do anything (no matter how harmless it is) before we validate it's authorized.

Reply:
I agree, good catch